### ISSUE TITLE
Tau Server Slow, Bad Frames

### ISSUE THREAD

**Issue by kevinboards** [OnionIoT/tau-lidar-camera#13]
This issue refers to this conversation
https://community.onion.io/topic/4509/tau-usb-cable-suggestions

Python Library versions
Python 3.9.5
pip 21.1.1

Camera Info
ToF Camera opened successfully
model: 4.0
firmware: 3.3
uid: 9500c6
resolution: 160x60
port: COM8

### Describe the bug
TauLidarServer.py runs slow. 
Lots of bad frames, new frame every 5 - 60+ seconds.  Sporadic.   

### Steps to Reproduce
Plug USB A to USB C cable into Tau Camera
in CMD type 'python -m TauLidarServer'

### What is expected?
Camera should take new frames at regular intervals
should not take more than a minute to get a frame.  I have seen some faster movement from the Hackaday article.  
Current speed would not be reliable to use in the field. 

### What is actually happening?
https://drive.google.com/file/d/1N6E0iDCx881zLVFrJZcKs9NFQYx86uRG/view?usp=sharing

Lots of bad Frames, Lots of time between frames.   In the video I have attached in a Google Drive link you will see the frames take a long time to update.   Every time the frames update I immediately change the cameras angle but it takes a long time for the camera to get a decent frame. 
Here is the CMD code from this video 

`C:\Users\Kevin>python -m TauLidarServer
Looking for connected Tau LiDAR Camera hardware ...

ToF camera opened successfully:
    model:      4.0
    firmware:   3.3
    uid:        9500C6
    resolution: 160x60
    port:       COM8
    IP address: 127.0.0.1
    URL:  http://127.0.0.1:8080

Press Ctrl + C keys to shutdown ...
127.0.0.1 - - [02/Jul/2021 23:12:57] "GET / HTTP/1.1" 200 -
127.0.0.1 - - [02/Jul/2021 23:12:57] "GET /css/spectre.min.css HTTP/1.1" 200 -
127.0.0.1 - - [02/Jul/2021 23:12:57] "GET /css/spectre-exp.min.css HTTP/1.1" 200 -
127.0.0.1 - - [02/Jul/2021 23:12:57] "GET /css/spectre-icons.min.css HTTP/1.1" 200 -
127.0.0.1 - - [02/Jul/2021 23:12:57] "GET /css/loading.css HTTP/1.1" 200 -
127.0.0.1 - - [02/Jul/2021 23:12:57] "GET /js/app.js HTTP/1.1" 200 -
127.0.0.1 - - [02/Jul/2021 23:12:57] "GET /img/tau-logo.svg HTTP/1.1" 200 -
127.0.0.1 - - [02/Jul/2021 23:12:57] "GET /js/vue.esm-browser.js HTTP/1.1" 200 -
127.0.0.1 - - [02/Jul/2021 23:12:57] "GET /js/pointsView.js HTTP/1.1" 200 -
127.0.0.1 - - [02/Jul/2021 23:12:58] "GET /js/three.module.js HTTP/1.1" 200 -
127.0.0.1 - - [02/Jul/2021 23:12:58] "GET /js/OrbitControls.js HTTP/1.1" 200 -
127.0.0.1 - - [02/Jul/2021 23:12:58] code 404, message File not found
127.0.0.1 - - [02/Jul/2021 23:12:58] "GET /favicon.ico HTTP/1.1" 404 -
Data error, actual size: 38356, expected size: 38480
Bad frame ignored, bytes length: 38276
skip frame
Data error, actual size: 38420, expected size: 38480
Bad frame ignored, bytes length: 38340
skip frame
Data error, actual size: 38356, expected size: 38480
Bad frame ignored, bytes length: 38276
skip frame
Data error, actual size: 38356, expected size: 38480
Bad frame ignored, bytes length: 38276
skip frame
Data error, actual size: 38356, expected size: 38480
Bad frame ignored, bytes length: 38276
skip frame
Data error, actual size: 38420, expected size: 38480
Bad frame ignored, bytes length: 38340
skip frame
Data error, actual size: 38356, expected size: 38480
Bad frame ignored, bytes length: 38276
skip frame
Data error, actual size: 38356, expected size: 38480
Bad frame ignored, bytes length: 38276
skip frame
Data error, actual size: 38356, expected size: 38480
Bad frame ignored, bytes length: 38276
skip frame
Data error, actual size: 38356, expected size: 38480
Bad frame ignored, bytes length: 38276
skip frame
Data error, actual size: 38164, expected size: 38480
Bad frame ignored, bytes length: 38084
skip frame
Data error, actual size: 38356, expected size: 38480
Bad frame ignored, bytes length: 38276
skip frame
Data error, actual size: 38228, expected size: 38480
Bad frame ignored, bytes length: 38148
skip frame
Data error, actual size: 38356, expected size: 38480
Bad frame ignored, bytes length: 38276
skip frame
code = 1006 (connection closed abnormally [internal]), no reason
code = 1006 (connection closed abnormally [internal]), no reason
Data error, actual size: 38356, expected size: 38480
Bad frame ignored, bytes length: 38276
skip frame
code = 1006 (connection closed abnormally [internal]), no reason
code = 1006 (connection closed abnormally [internal]), no reason
code = 1006 (connection closed abnormally [internal]), no reason

Shutting down ...`

**Comment by greenbreakfast** [comment]
@kevinboards thanks for posting an issue, we'll do our best to sort this out.

A few questions from our end:
* Does this happen every time you try to run TauLidarServer? 
* Has it worked as expected before? If so, did it consistently work as expected or only from time to time?
* Do you have multiple Tau Cameras, and if so, does this happen with all of them?

**Comment by kevinboards** [comment]
> @kevinboards thanks for posting an issue, we'll do our best to sort this out.
> * Does this happen every time you try to run TauLidarServer?  
Yes
> * Has it worked as expected before? If so, did it consistently work as expected or only from time to time?
No, it has never worked as expected. 
> * Do you have multiple Tau Cameras, and if so, does this happen with all of them?
I only have 1 camera

**Comment by kevinboards** [comment]
UPDATE: Today I was able to get my hands on an old windows 7 machine.   I downloaded Python 3.7, setup everything for the TauLidarServer and it ran  really well!  No problems.  
I then downloaded the examples, the recording of frames was incredibly fast and all of my frames were good frames.    

The Tau Camera is obviously fine.  There is something going on with my lenovo laptop with python 3.9 that I will need to get to the bottom of.
